### PR TITLE
Spliced together parts of old and new versions of the code.

### PR DESCRIPTION
**Pull request by mullinska**
List of changes made:
Specifying filetypes in function declarations
Swapped back to the Fit_Method class instead of the FittingMethod class (up for discussion)
Directory the code is called from changed back to fit_resonator instead of measurement
Keeping more in-depth error messages
Sticking with VNASweep class to input data for the code
Fixed error where np.append for linregress had hard coded 10 instead of using normalize variable
Added ability for user to specify if they want to print the normalization process plots or not
Stopped usage of useless functions in Resonator class file
Updated requirements.txt
Re-implemented plotting, monte carlo fit, least squares fit, confidence intervals, and initial guess functions
Re-implemented phi rotation method
Removed find_circle boolean in Method class (wasn't being used at all)
Updated extract_near_res function
Specified that frequency is in GHz
Fixed newline error for title of plot
Removed capitals from function names to fit with conventions

### CODE REVIEW

**Comment by crmcrae** [comment]
This looks good to me. It would be ideal if @joshmutus could sign off on this as many of the modifications are to code that he initiated.

File: resfit/fit_resonator/fit_S_data.py [hunk 1]
```diff
@@ -31,15 +40,664 @@
...
+        if Method.method == 'INV':
+            phi = np.angle(z_c)  ###########4
+        else:
+            phi = np.angle(-z_c)  ###########4
+
+        freq_idx = np.argmax(np.abs(ydata))
+        f_c = x[freq_idx] ###############3
+
+        if plot_extra:
+            plot(np.real(ydata),np.imag(ydata),"resonance",output_path,np.real(z_c),np.imag(z_c),r,np.real(ydata[freq_idx]),np.imag(ydata[freq_idx]))#plot data with guide circle
```
Comment (joshmutus) [inline resfit/fit_resonator/fit_S_data.py]: Please break this up over mulitple lines like:
```python
    if plot_extra:       
        plot(np.real(ydata),
             np.imag(ydata),
             "resonance",
             output_path,
             np.real(z_c),
             np.imag(z_c),
             r,
             np.real(ydata[freq_idx]),
             np.imag(ydata[freq_idx]))
```

File: resfit/fit_resonator/fit_S_data.py [hunk 2]
```diff
@@ -31,15 +40,664 @@
...
+        if f_c < 0:
+            print(">Resonance frequency is negative. Please only input positive frequencies.")
+            quit()
+    except:
+        print(">Cannot find resonance frequency in Find_initial_guess")
+        quit()
+
+    if Method.method == 'DCM' or Method.method == 'PHI':
+        try:
+            Q_Qc = np.max(np.abs(ydata)) #diameter of the circle found from getting distance from (0,0) to resonance frequency data point (possibly should be using fit circle)
```
Comment (joshmutus) [inline resfit/fit_resonator/fit_S_data.py]: Please put the comment on line before `Q_Qc = ....` so it's easier to read

File: resfit/fit_resonator/fit_S_data.py [hunk 3]
```diff
@@ -31,15 +40,664 @@
...
+
+########################################################################
+def find_nearest(array,value):
+    idx = (np.abs(array-value)).argmin()
+    val = array[idx]
+    return val, idx
+
+###########################################################
+   ## PLOT REAL and IMAG figure
+def PlotFit(x,y,x_initial,y_initial,slope,intercept,slope2,intercept2,params,Method,func,error,figurename,x_c,y_c,radius,output_path,conf_array,extract_factor = None,title = "Fit",manual_params = None):
```
Comment (joshmutus) [inline resfit/fit_resonator/fit_S_data.py]: please break up arguments over muliple lines, and rename to `plot_fit`

File: README.md [hunk 4]
```diff
@@ -8,32 +8,7 @@ Welcome to the measurement repostository of the Boulder Cryogenic Quantum Testbe
...
-method = ff.FittingMethod.DCM
-
-# Fit the data:
-fsd.fit_resonator(raw, guess, method, 10, None)
->>>{'Q': 273178.7431584903,
- 'Qc': 242764.2007247585,
- 'w1': 5.78529287158879,
- 'phi': 0.06757726597956264}
-```
+See the resfit folder for how to use the fitting code.
```
Comment (joshmutus) [inline README.md]: So can you write a short example that users can run the code (say with one of the datasets in the repo) and the example expected output. We need a very simple example users can run to convince themselves the code works when they install it.

**Pull request closed without merging**